File: integrations/src/main/java/com/nivasafinance/integrations/framework/core/service/ThirdPartyProviderConfigReadService.java
package com.nivasafinance.integrations.framework.core.service;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.nivasafinance.integrations.framework.core.data.ThirdPartyConfig;
import com.nivasafinance.integrations.framework.core.exception.ServiceConfigurationException;
import com.nivasafinance.integrations.framework.core.repository.ThirdPartyProviderConfigRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

@Service
public class ThirdPartyProviderConfigReadService {

    private final ThirdPartyProviderConfigRepository thirdPartyProviderConfigRepository;
    private final Gson gson = new Gson();

    @Autowired
    public ThirdPartyProviderConfigReadService(ThirdPartyProviderConfigRepository thirdPartyProviderConfigRepository) {
        this.thirdPartyProviderConfigRepository = thirdPartyProviderConfigRepository;
    }

    public ThirdPartyConfig getProviderConfigById(UUID id) {
        var configEntity = thirdPartyProviderConfigRepository.findById(id)
                .orElseThrow(() -> new ServiceConfigurationException("Config not found: " + id));
        
        if (!configEntity.getActive()) {
            throw new ServiceConfigurationException("Config " + id + " is not active");
        }
        
        Type type = new TypeToken<HashMap<String, String>>() {}.getType();
        Map<String, String> configurations = gson.fromJson(configEntity.getConfigs(), type);
        
        return new ThirdPartyConfig(
                configEntity.getId(),
                configEntity.getName(),
                configEntity.getProvider(),
                configurations
        );
    }
}

File: integrations/src/main/kotlin/framework/core/service/ThirdPartyProviderConfigReadService.kt
package framework.core.service

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import framework.core.data.ThirdPartyConfig
import framework.core.exception.ServiceConfigurationException
import framework.core.repository.ThirdPartyProviderConfigRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class ThirdPartyProviderConfigReadService(val thirdPartyProviderConfigRepository: ThirdPartyProviderConfigRepository) {

    fun getProviderConfigById(id: UUID): ThirdPartyConfig {
        val configEntity = thirdPartyProviderConfigRepository.findById(id).orElseThrow()
        if (!configEntity.active) {
            throw ServiceConfigurationException("Config $id is not active")
        }
        val configurations = Gson().fromJson<MutableMap<String, String>>(
            configEntity.configs,
            object : TypeToken<HashMap<String, String>>() {}.type
        ); // todo
        return ThirdPartyConfig(
            id = configEntity.id,
            provider = configEntity.provider,
            name = configEntity.name,
            configurations = configurations
        )
    }
}
